### PR TITLE
partial refactor as discussed in #1363

### PR DESCRIPTION
**Pull request by n-hutton**

### CODE REVIEW

File: libs/dkg/include/dkg/rbc.hpp [hunk 1]
```diff
@@ -83,22 +79,19 @@ class RBC
   struct Party
   {
     std::unordered_map<TagType, std::bitset<sizeof(MsgType) * 8>>
-                                    flags;  ///< Marks for each message tag what messages have been received
-    uint8_t                         deliver_s;        ///< Counter for messages delivered
+            flags;          ///< Marks for each message tag what messages have been received
+    uint8_t deliver_s = 1;  ///< Counter for messages delivered - initialised to 1
```
Comment (wkaluza) [inline libs/dkg/include/dkg/rbc.hpp:83]: why 1?

File: libs/dkg/src/dkg_service.cpp [hunk 2]
```diff
@@ -439,7 +439,7 @@ State DkgService::OnCollectSignaturesState()
         // verify the signature
         if (!crypto::bls::Verify(it->signature, it->public_key, payload))
         {
-          FETCH_LOG_ERROR(LOGGING_NAME, "Failed to very signature submision. Discarding");
+          FETCH_LOG_ERROR(LOGGING_NAME, "Failed to very signature submission. Discarding");
```
Comment (wkaluza) [inline libs/dkg/src/dkg_service.cpp:442]: typo: Failed to verify

File: libs/dkg/src/rbc.cpp [hunk 3]
```diff
@@ -145,7 +145,7 @@ void RBC::Broadcast(const RBCEnvelop &env)
 }
 
 /**
- * Places a serialised message to be broadcasted via RBC into a RBCEnvelop and broadcats
+ * Places a serialised message to be broadcasted via RBC into a RBCEnvelop and broadcasts
```
Comment (wkaluza) [inline libs/dkg/src/rbc.cpp:148]: `RBCEnvelope`? `envelope` in the line below?

File: libs/crypto/include/fetch_bls.hpp [hunk 4]
```diff
@@ -27,6 +27,7 @@
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wconversion"
 #pragma clang diagnostic ignored "-Wpedantic"
+#pragma clang diagnostic ignored "-Wmacro-redefined"
```
Comment (wkaluza) [inline libs/crypto/include/fetch_bls.hpp:30]: It's not just clang that moans about this

**Review by wkaluza — CHANGES_REQUESTED** [review]

File: libs/dkg/src/dkg_service.cpp (hunk 2 above)
Comment (n-hutton) [reply libs/dkg/src/dkg_service.cpp:442]: yup

File: libs/dkg/src/rbc.cpp (hunk 3 above)
Comment (n-hutton) [reply libs/dkg/src/rbc.cpp:148]: I believe that envelop should become envelope but it's not a priority

File: libs/crypto/include/fetch_bls.hpp (hunk 4 above)
Comment (n-hutton) [reply libs/crypto/include/fetch_bls.hpp:30]: gotcha

File: libs/dkg/include/dkg/rbc.hpp (hunk 1 above)
Comment (n-hutton) [reply libs/dkg/include/dkg/rbc.hpp:83]: It seems it's a feature of the RBC protocol

**Review by jinmannwong — APPROVED** [review]
Looks good.

**Pull request merged**